assets/blocks/quiz/quiz-block/use-has-questions.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+import { select, dispatch } from '@wordpress/data';
+
+/**
+ * Monitor for questions and disable the lesson quiz when none have been added.
+ *
+ * @param {string} clientId The quiz block client id.
+ */
+export const useHasQuestions = ( clientId ) => {
+	const questionBlocks = select( 'core/block-editor' )
+		.getBlocks( clientId )
+		.filter( ( block ) => !! block.attributes.title );
+
+	// Monitor for valid questions.
+	useEffect( () => {
+		const quizHasQuestions = select( 'core/editor' ).getEditedPostAttribute(
+			'meta'
+		)?._quiz_has_questions; // eslint-disable-line camelcase
```

Comment:
I don't think the Eslint comment is needed — it doesn't complain for me without it.
Can also use a destructuring trick:

```suggestion
		const { _quiz_has_questions: quizHasQuestions } =
			select( 'core/editor' ).getEditedPostAttribute( 'meta' ) || {};
```

Reply:
Looks good. Copied this from our `_needs_template` code.